src/clif-util.rs

```diff
@@ -45,125 +41,179 @@ mod print_cfg;
+                .arg(add_input_file_arg()),
+        )
+        .subcommand(
+            add_wasm_or_compile("compile")
+                .arg(
+                    Arg::with_name("just_decode")
+                        .short("t")
+                        .help("Just decode WebAssembly to Cranelift IR"),
+                )
+                .arg(Arg::with_name("check_translation").short("c").help(
+                    "Just checks the correctness of Cranelift IR translated from WebAssembly",
+                )),
+        )
+        .subcommand(add_wasm_or_compile("wasm"));
+
+    let res_util = match app_cmds.get_matches().subcommand() {
+        ("cat", Some(rest_cmd)) => {
+            let mut file_vec: Vec<String> = Vec::new();
+            get_vec(&mut file_vec, rest_cmd.values_of("file"));
+            cat::run(&file_vec)
+        }
+        ("test", Some(rest_cmd)) => {
+            let mut file_vec: Vec<String> = Vec::new();
+            get_vec(&mut file_vec, rest_cmd.values_of("file"));
+            cranelift_filetests::run(rest_cmd.is_present("time-passes"), &file_vec).map(|_time| ())
+        }
+        ("filecheck", Some(rest_cmd)) => {
+            let mut file_vec: Vec<String> = Vec::new();
+            get_vec(&mut file_vec, rest_cmd.values_of("file"));
+            rsfilecheck::run(&file_vec, rest_cmd.is_present("verbose"))
+        }
+        ("print-cfg", Some(rest_cmd)) => {
+            let mut file_vec: Vec<String> = Vec::new();
+            get_vec(&mut file_vec, rest_cmd.values_of("file"));
+            print_cfg::run(&file_vec)
+        }
+        ("compile", Some(rest_cmd)) => {
+            let mut file_vec: Vec<String> = Vec::new();
+            get_vec(&mut file_vec, rest_cmd.values_of("file"));
+
+            let mut set_vec: Vec<String> = Vec::new();
+            get_vec(&mut set_vec, rest_cmd.values_of("set"));
+
+            let mut target_val: &str = "";
+            if let Some(clap_target_vec) = rest_cmd.values_of("target") {
+                for val in clap_target_vec {
+                    target_val = val;
+                }
+            }
+            compile::run(file_vec, rest_cmd.is_present("print"), &set_vec, target_val)
+        }
+        ("wasm", Some(rest_cmd)) => {
+            let mut file_vec: Vec<String> = Vec::new();
+            get_vec(&mut file_vec, rest_cmd.values_of("file"));
+
+            let mut set_vec: Vec<String> = Vec::new();
+            get_vec(&mut set_vec, rest_cmd.values_of("set"));
+
+            let mut target_val: &str = "";
+            if let Some(clap_target_vec) = rest_cmd.values_of("target") {
+                for val in clap_target_vec {
+                    target_val = val;
+                }
+            }
+
+            #[cfg(feature = "wasm")]
+            let result = wasm::run(
+                file_vec,
+                rest_cmd.is_present("verbose"),
+                rest_cmd.is_present("just-decode"),
+                rest_cmd.is_present("check-translation"),
```

Comment:
Above, the arguments are named `just_decode` and `check_translation`, with underscores, but the names here have dashes. Are these supposed to match those?